Document hotfix process (#701)
openstack-k8s-operators/dataplane-operator: docs/assemblies/proc_hotfixing-the-data-plane-rpm-content.adoc

[id="proc_hotfixing-the-data-plane-rpm-content-{context}"]
= Hotfixing the data plane RPM content

[role="_abstract"]

> RPM hotfix content is installed directly on to the data plane nodes.

Reviewer: ```suggestion
You install RPM hotfix content directly on to the data plane nodes.
```

.Procedure

> . Obtain the RPM hotfix content from the source and store it locally.

Reviewer: ```suggestion
. Obtain the RPM hotfix content from the source and store it locally:
```

+
----
$ mkdir -p <hotfix_id>/rpms
$ cp /path/to/hotfix/*.rpm <hotfix_id>/rpms
----
+

> * Replace `<hotfix_id>` with a hotfix identifier such as a Jira issue, for example `osprh-0000`

Reviewer: ```suggestion
* Replace `<hotfix_id>` with a hotfix identifier such as a Jira issue, for example `osprh-0000`.
```

> . Copy the RPM hotfix content to the affected data plane nodes.

Reviewer: ```suggestion
. Copy the RPM hotfix content to the affected data plane nodes:
```

+
----
$ ssh <ssh_user>@<data_plane_node> mkdir -p /tmp/<hotfix_id>/rpms
$ scp <hotfix_id>/rpms/*.rpm <ssh_user>@<data_plane_node>:/tmp/<hotfix_id>/rpms
----
+

> * Replace `<hotfix_id>` with a hotfix identifier such as a Jira issue, for example `osprh-0000`

Reviewer: Move the <hotfix_id> replace statement to after the <data_plane_node> statement, so that the user-replaceable variables are listed in the same order as they appear in the commands.

Author: Done.

> * Replace `<ssh_user>` with the user name used for SSH

Reviewer: ```suggestion
* Replace `<ssh_user>` with the SSH user name.
```

* Replace `<data_plane_node>` with the hostname or IP for the data plane node.

> * Repeat the command for each affected data plane node.

Reviewer: ```suggestion

+
Repeat this step for each data plane node that the hotfix must be applied to.
```
Using a * makes this statement appear in the list of replace statements. I think you want it to stand out on its own as a "Repeat this step". The suggested syntax should make the statement appear at the end of the step. If it skews the output, try taking the line space out from before the +.

I've also suggested changing the statement to read as "repeat the step", as there is more than one command in the instruction.

> . Update the RPM hotfix content on the affected data plane nodes.

Author: By "affected", I mean the nodes where the hotfix needs to be applied. It depends on the actual content of the hotfix and where the previous version of the software is already applied.

. Update the RPM hotfix content on the affected data plane nodes.
+
----
$ ssh <ssh_user>@<data_plane_node>

> $ sudo rpm -F /tmp/hotfix-rpms/*.rpm

Reviewer 2: This diffent dir path, then in scp command above.

Author: fixed.

> * The `-F` (`--freshen`) `rpm` command option will update the RPM content if earlier versions are already installed.

Reviewer: ```suggestion
* The `-F` (`--freshen`) `rpm` command option updates the RPM content if earlier versions are already installed.
```
Avoid future tense.

> . Complete any remaining custom steps to apply the RPM hotfix content
> +
> In some cases, custom steps may be required to complete applying the hotfix content. This may include restarting `systemd` services or other steps. Refer to the instructions that accompanied the hotifx and complete any necessary steps.

Reviewer: ```suggestion
. Perform any additional custom steps that are detailed in the hotfix instructions to complete the application of the RPM hotfix content. 
```
